extensions/omni_httpd/master_worker.c

```diff
@@ -511,12 +511,15 @@ void master_worker(Datum db_oid) {
       strncpy(worker.bgw_library_name, MyBgworkerEntry->bgw_library_name, BGW_MAXLEN - 1);
       for (int i = 0; i < *num_http_workers; i++) {
         BackgroundWorkerHandle *handle;
-        RegisterDynamicBackgroundWorker(&worker, &handle);
-        pid_t worker_pid;
-        if (WaitForBackgroundWorkerStartup(handle, &worker_pid) == BGWH_POSTMASTER_DIED) {
-          goto terminate;
+        if (RegisterDynamicBackgroundWorker(&worker, &handle)) {
+          pid_t worker_pid;
+          if (WaitForBackgroundWorkerStartup(handle, &worker_pid) == BGWH_POSTMASTER_DIED) {
+            goto terminate;
+          }
+          cvec_bgwhandle_push(&http_workers, handle);
+        } else {
+          ereport(WARNING, errmsg("Could not start %s", worker.bgw_name));
```

Comment:
Could the `RegisterDynamicBackgroundWorker` fail for other cases except the `max_worker_processes` one?

If not, it would be good for operators to put a HINT here, like:
```
HINT: Ensure omni_httpd.http_workers doesn't surpass max_worker_processes
```

Reply:
It "**typically** indicates that no background worker slots are currently available" (from the documentation)

Other cases that can be immediately seen:

* Executed in postmaster. Can't happen here.
* "If this is a parallel worker, check whether there are already too many parallel workers; if so, don't register another one.". This is not a parallel worker


So far, it seems that it is fairly plausible that this would be the only case and HINT would be appropriate.

Reply:
Done. The message now looks like:

```
WARNING:  Could not start omni_httpd worker
DETAIL:  omni_httpd.http_workers = 16 is higher than max_worker_processes = 3
HINT:  Consider increasing max_worker_processes
```

It might seem like bikeshedding but from what I've seen good error messages help a lot. I think they fit in the D3X concept too.

Reply:
Don't think this is bikeshedding. Good errors are important. I also left some minor commentary there, we can look at it now or as a follow-up improvement PR.